service/notesservice.php

```diff
@@ -262,6 +280,19 @@ private function generateFileName (Folder $folder, $title, $extension, $id) {
         }
     }
 
+    private function getCategoryPath($category) {
+        $category = str_replace('\\', '/', $category);
+        $folders = explode('/', $category);
+        $cleanedFolders = [];
+        foreach($folders as $folder) {
+            $folder = trim($folder);
+            $folder = preg_replace('/^\.+\s*/', '', $folder);
+            if(!empty($folder)) {
+                $cleanedFolders[] = $folder;
+            }
+        }
+        return implode('/', $cleanedFolders);
+    }
```

Comment:
Okay, I've removed the category name preparation and introduced an exception handling. Please review again, @Henni